tst/EnergyPlus/unit/CurveManager.unit.cc

```diff
@@ -742,3 +744,233 @@ TEST_F(EnergyPlusFixture, CSV_CarriageReturns_Handling)
+    }
+    EXPECT_EQ(6, unit_type_choices.size());
+}
+
+class InputUnitTypeIsValid : public EnergyPlusFixture, public ::testing::WithParamInterface<std::string_view>
+{
+};
+TEST_P(InputUnitTypeIsValid, IndepentVariable)
+{
+    const auto &unit_type = GetParam();
+
+    std::string const idf_objects = delimited_string({
+        "Table:IndependentVariable,",
+        "  SAFlow,                    !- Name",
+        "  Cubic,                     !- Interpolation Method",
+        "  Constant,                  !- Extrapolation Method",
+        "  0.714,                     !- Minimum Value",
+        "  1.2857,                    !- Maximum Value",
+        "  ,                          !- Normalization Reference Value",
+        fmt::format("  {},             !-  Unit Type", unit_type),
+        "  ,                          !- External File Name",
+        "  ,                          !- External File Column Number",
+        "  ,                          !- External File Starting Row Number",
+        "  0.714286,                  !- Value 1",
+        "  1.0,",
+        "  1.2857;",
+
+        "Table:IndependentVariableList,",
+        "  SAFlow_Variables,          !- Name",
+        "  SAFlow;                    !- Independent Variable 1 Name",
+
+        "Table:Lookup,",
+        "  CoolCapModFuncOfSAFlow,    !- Name",
+        "  SAFlow_Variables,          !- Independent Variable List Name",
+        "  ,                          !- Normalization Method",
+        "  ,                          !- Normalization Divisor",
+        "  0.8234,                    !- Minimum Output",
+        "  1.1256,                    !- Maximum Output",
+        "  Dimensionless,             !- Output Unit Type",
+        "  ,                          !- External File Name",
+        "  ,                          !- External File Column Number",
+        "  ,                          !- External File Starting Row Number",
+        "  0.823403,                  !- Output Value 1",
+        "  1.0,",
+        "  1.1256;",
+    });
+
+    ASSERT_TRUE(process_idf(idf_objects));
+    EXPECT_EQ(0, state->dataCurveManager->NumCurves);
+
+    Curve::GetCurveInput(*state);
+    state->dataCurveManager->GetCurvesInputFlag = false;
+    EXPECT_TRUE(compare_err_stream("", true));
+}
+
+INSTANTIATE_TEST_SUITE_P(CurveManager,
+                         InputUnitTypeIsValid,
+                         testing::Values("", "Angle", "Dimensionless", "Distance", "MassFlow", "Power", "Temperature", "VolumetricFlow"),
+                         [](const testing::TestParamInfo<InputUnitTypeIsValid::ParamType> &info) -> std::string {
+                             if (info.param.empty()) {
+                                 return "Blank";
+                             }
+                             return std::string{info.param};
+                         });
```

Comment:
Instantiate the param tests... I could have used testing::ValuesIn(container) instead to make it fully dynamic but didn't